CMakeLists.txt

```diff
@@ -793,7 +793,11 @@ endif()
 option(ENABLE_UNIX_SOCKETS "Define if you want Unix domain sockets support" ON)
 if(ENABLE_UNIX_SOCKETS)
   include(CheckStructHasMember)
-  check_struct_has_member("struct sockaddr_un" sun_path "sys/un.h" USE_UNIX_SOCKETS)
+  if(WIN32)
+    set(USE_UNIX_SOCKETS ON)
```

Comment:
This assumes unix sockets is always available on windows?

Reply:
Yes.
With this PR, whether msvc or mingw, whether newer verison or older version, compilation will always pass when unix sockets feature is ON.

Reply:
Ok but do you know unix sockets to always be available on Windows? AFAIK it is only in the most recent builds of Windows.

/cc @vszakats

Reply:
Although unix sockets support on Windows is introduced by Windows 10 SDK 17063. But actually, if we add the definition of sockaddr_un structure manually, it can be extended to all build toolchains on Windows.

See also: https://github.com/curl/curl/commit/23a870f2fd041278762ecf819cd1467019588c58

Reply:
A Windows binary that supports UnixSockets can be built regardless of the toolchain or SDK version. As to whether UnixSockets works or not will/may depend on the actual Windows environment the binary is executed in. This patch brings nothing new here, only extends extisting support to CMake.